With v3.0.1, getting error "an unknown tool name found in scoped lint"

I am using rust 1.56.1 and after doing a `cargo  update` on my project the `fd-lock` version changed from v03.0.0 to v3.0.1 (as a transitive dependency).  I now get the below error: 
```
...
Compiling fd-lock v3.0.1
error[E0710]: an unknown tool name found in scoped lint: `rustdoc::missing_doc_code_examples`
  --> /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/fd-lock-3.0.1/src/lib.rs:33:23
   |
33 | #![warn(missing_docs, rustdoc::missing_doc_code_examples)]
   |                       ^^^^^^^
```
I believe this is because `rustdoc::missing_doc_code_example` is only available in `nightly`. I worked around the issue by pinning the version of `fd-lock` to `3.0.0` in my project, but I believe your MSRV is stable so I shouldn't get this error.

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -30,7 +30,7 @@
 
 #![forbid(future_incompatible)]
 #![deny(missing_debug_implementations, nonstandard_style)]
-#![warn(missing_docs, rustdoc::missing_doc_code_examples)]
+#![cfg_attr(doc, warn(missing_docs, rustdoc::missing_doc_code_examples))]
 
 mod read_guard;
 mod rw_lock;

